File: src/test/java/com/lucas/matches/service/converter/MatchConverterTest.java
package com.lucas.matches.service.converter;

import com.lucas.matches.infrastructure.document.MatchDocument;
import com.lucas.matches.infrastructure.document.SummaryType;
import com.lucas.matches.service.domain.Match;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;

import java.time.LocalDateTime;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
public class MatchConverterTest {
    private MatchDocument avbMatchDocument;
    private MatchDocument avbTimeMatchDocument;
    private MatchDocument noSummaryTypeMatchDocument;
    private MatchConverter matchConverter;

    @BeforeAll
    public void setup(){
        avbMatchDocument = new MatchDocument("1", "t1", "m1", "playerA", "playerB", LocalDateTime.now(), SummaryType.AVB);
        avbTimeMatchDocument = new MatchDocument("1", "t1", "m1", "playerA", "playerB", LocalDateTime.now(), SummaryType.AVBTIME);
        noSummaryTypeMatchDocument = new MatchDocument("1", "t1", "m1", "playerA", "playerB", LocalDateTime.now(), null);
        matchConverter = new MatchConverter();
    }

    @Test
    @DisplayName("convert MatchDocument with AVB SummaryType to Match")
    public void convertMatchDocumentAvbToMatch(){
        List<Match> match = matchConverter.convert(List.of(avbMatchDocument));
        assertNotNull(match);
        assertEquals(1, match.size());
        assertEquals(com.lucas.matches.service.domain.SummaryType.AVB, match.get(0).getSummaryType());
    }

    @Test
    @DisplayName("convert MatchDocument with AVBTime SummaryType to Match")
    public void convertMatchDocumentAvbTimeToMatch(){
        List<Match> match = matchConverter.convert(List.of(avbTimeMatchDocument));
        assertNotNull(match);
        assertEquals(1, match.size());
        assertEquals(com.lucas.matches.service.domain.SummaryType.AVBTIME, match.get(0).getSummaryType());
    }
    @Test
    @DisplayName("convert MatchDocument with no SummaryType to Match")
    public void convertMatchDocumentNoSummaryTypeToMatch(){
        List<Match> match = matchConverter.convert(List.of(noSummaryTypeMatchDocument));
        assertNotNull(match);
        assertEquals(1, match.size());
        assertNull(match.get(0).getSummaryType());
    }
}

File: src/test/kotlin/com/lucas/matches/service/converter/MatchConverterTest.kt
package com.lucas.matches.service.converter

import com.lucas.matches.infrastructure.document.MatchDocument
import com.lucas.matches.infrastructure.document.SummaryTypeDocument
import com.lucas.matches.service.domain.SummaryType
import org.junit.jupiter.api.*
import java.time.LocalDateTime

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class MatchConverterTest {
    private lateinit var avbMatchDocument: MatchDocument
    private lateinit var avbTimeMatchDocument: MatchDocument
    private lateinit var noSummaryTypeMatchDocument: MatchDocument
    private lateinit var matchConverter: MatchConverter
    @BeforeAll
    fun setup() {
        avbMatchDocument =
            MatchDocument("1", "t1", "m1", "playerA", "playerB", LocalDateTime.now(), SummaryTypeDocument.AVB)
        avbTimeMatchDocument =
            MatchDocument("1", "t1", "m1", "playerA", "playerB", LocalDateTime.now(), SummaryTypeDocument.AVBTIME)
        noSummaryTypeMatchDocument = MatchDocument("1", "t1", "m1", "playerA", "playerB", LocalDateTime.now(), null)
        matchConverter = MatchConverter()
    }

    @Test
    @DisplayName("convert MatchDocument with AVB SummaryType to Match")
    fun convertMatchDocumentAvbToMatch() {
        val match = matchConverter.convert(listOf(avbMatchDocument))
        Assertions.assertNotNull(match)
        Assertions.assertEquals(1, match.size)
        Assertions.assertEquals(SummaryType.AVB, match[0].summaryType)
    }

    @Test
    @DisplayName("convert MatchDocument with AVBTime SummaryType to Match")
    fun convertMatchDocumentAvbTimeToMatch() {
        val match = matchConverter.convert(listOf(avbTimeMatchDocument))
        Assertions.assertNotNull(match)
        Assertions.assertEquals(1, match.size)
        Assertions.assertEquals(SummaryType.AVBTIME, match[0].summaryType)
    }

    @Test
    @DisplayName("convert MatchDocument with no SummaryType to Match")
    fun convertMatchDocumentNoSummaryTypeToMatch() {
        val match = matchConverter.convert(listOf(noSummaryTypeMatchDocument))
        Assertions.assertNotNull(match)
        Assertions.assertEquals(1, match.size)
        Assertions.assertNull(match[0].summaryType)
    }

    @Test
    @DisplayName("convert empty List")
    fun convertEmptyList() {
        val match = matchConverter.convert(ArrayList())
        Assertions.assertNotNull(match)
        Assertions.assertEquals(0, match.size)
    }
}